Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3112,6 +3168,39 @@ public void run() {
 		});
 	}
 
+	private void processRedeemRewardsResponse(final ServerResponse resp) {
+		boolean updateListener = false;
```

Comment:
Also, `updateListener` and `finUpdateListener` aren't good variable names, try using something more descriptive like `redemptionSucceeded`.